src/lib/lib.jl

```diff
@@ -43,6 +43,32 @@ unwrap(x) = x
 
 @adjoint unwrap(x) = unwrap(x), Δ ->(accum_param(__context__, x, Δ); (Δ,))
 
+# map
+
+# TODO: Replace with more performant version. See https://github.com/JuliaLang/julia/issues/13942.
+unzip(zipped) = collect(zip(zipped...))
+
+function sum_named_tuple(tpl)
+    if tpl[1] == nothing
+        return nothing
+    end
+    vals = values.(tpl)
+    summed_vals = reduce((a, b) -> a.+b, vals[1:end-1]; init=vals[end])
+    NamedTuple{keys(tpl[1])}(summed_vals)
+end
+
+@adjoint function map(f, args...)
+  ys_and_backs = map((args...) -> Zygote._forward(f, args...), args...)
+  ys, backs = unzip(ys_and_backs)
```

Comment:
Unzip is pretty slow, in this case you could just do `ys = map(x -> x[1], ys_and_backs` etc.